### PR TITLE
Add js-class-names rule

### LINKED ISSUE #4: Lint static literal usage of js- class names
`js-foo` should be okay. `js-foo-${bar}` should be a lint error.

### PR DESCRIPTION
**Pull request by josh**
Closes #4.

-
CC: @mislav @dgraham

### CODE REVIEW

File: lib/rules/js-class-name.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  return {
+    "Literal": function(node) {
+      if (typeof node.value === 'string') {
+        checkStringFormat(node, node.value)
+
+        if (node.parent &&
+            node.parent.type === 'BinaryExpression' &&
+            node.parent.operator === '+' &&
```
Comment (mislav) [inline lib/rules/js-class-name.js]: What's with the plus operator?

File: lib/rules/js-class-name.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    var matches = str.match(allJsClassNameRegexp) || []
+    matches.forEach(function(match) {
+      if (!match.match(validJsClassNameRegexp)) {
+        context.report(node, "js- class names should be lowercase and only contain dashes.")
+      }
+    })
+  }
+
+  function checkStringEndsWithJSClassName(node, str) {
+    if (str.match(endWithJsClassNameRegexp)) {
```
Comment (mislav) [inline lib/rules/js-class-name.js]: Would `str.endsWith(allJsClassNameRegexp)` work?

File: lib/rules/js-class-name.js (hunk 1 above)
Comment (josh) [inline lib/rules/js-class-name.js]: Looking for `"foo" + bar`

File: lib/rules/js-class-name.js (hunk 2 above)
Comment (josh) [inline lib/rules/js-class-name.js]: I *think* `endsWith` only accepts strings, not a regexp.

**Pull request merged**